components/region_cache_memory_engine/src/range_manager.rs

```diff
@@ -0,0 +1,260 @@
+// Copyright 2024 TiKV Project Authors. Licensed under Apache-2.0.
+
+use std::collections::{BTreeMap, BTreeSet};
+
+use engine_traits::CacheRange;
+
+use crate::engine::{RagneCacheSnapshotMeta, SnapshotList};
+
+#[derive(Debug, Default)]
+pub struct RangeMeta {
+    id: u64,
+    range_snapshot_list: SnapshotList,
+    evicted: bool,
+    can_read: bool,
+    safe_ts: u64,
+}
+
+impl RangeMeta {
+    fn new(id: u64) -> Self {
+        Self {
+            id,
+            range_snapshot_list: SnapshotList::default(),
+            evicted: false,
+            can_read: false,
+            safe_ts: 0,
```

Comment:
we probably need to change the `safe_ts` to `safe_point` to be consistent with current tikv terms. 
Because today we use `safe_point` as the max timestamp to gc in compaction filter and use `safe_ts` as the min timestamp of all data are applied.